core/internal/leet/leftsidebar.go

```diff
@@ -383,16 +383,29 @@
 	maxKeyWidth := int(float64(width) * leftSidebarKeyWidthRatio)
 	maxValueWidth := width - maxKeyWidth - 1
 
+	itemCount := len(section.FilteredItems)
+	if itemCount == 0 || section.ItemsPerPage <= 0 {
+		return nil
+	}
+
 	startIdx := section.CurrentPage * section.ItemsPerPage
-	endIdx := min(startIdx+section.ItemsPerPage, len(section.FilteredItems))
+	endIdx := min(startIdx+section.ItemsPerPage, itemCount)
 	actualItemsToShow := endIdx - startIdx
+	if actualItemsToShow <= 0 {
+		// Out‑of‑range page; with normalizePagination this shouldn't happen,
+		// but defensively avoid negative capacities.
+		return nil
+	}
 
 	itemsToRender := min(actualItemsToShow, section.ItemsPerPage)
+	if itemsToRender <= 0 {
+		return nil
+	}
```

Comment:
All these checks are redundant with the `range` statement, which [won't loop at all](https://go.dev/ref/spec#RangeClause:~:text=If%20n%20%3C%3D%200%2C%20the%20loop%20does%20not%20run%20any%20iterations.) if `itemsToRender <= 0`. I'm guessing that a negative capacity on the line initializing `lines` would panic, but for that you can add a `max(0, ...)` to the definition of `itemsToRender`.